libraries/SPI/SPI.cpp

```diff
@@ -234,23 +227,158 @@ uint16_t ArduinoSPI::transfer16(uint16_t data)
 
 void ArduinoSPI::transfer(void *buf, size_t count)
 {
-  _spi_cb_event[_cb_event_idx] = SPI_EVENT_TRANSFER_ABORTED;
+    if (_is_sci) {
+        _spi_cb_event[_cb_event_idx] = SPI_EVENT_TRANSFER_ABORTED;
 
-  if (_is_sci) {
-    _write_then_read(&_spi_sci_ctrl, buf, buf, count, SPI_BIT_WIDTH_8_BITS);
-  } else {
-    _write_then_read(&_spi_ctrl, buf, buf, count, SPI_BIT_WIDTH_8_BITS);
-  }
+        _write_then_read(&_spi_sci_ctrl, buf, buf, count, SPI_BIT_WIDTH_8_BITS);
 
-  for (auto const start = millis();
-       (SPI_EVENT_TRANSFER_COMPLETE != _spi_cb_event[_cb_event_idx]) && (millis() - start < 1000); )
-  {
-      __NOP();
-  }
-  if (SPI_EVENT_TRANSFER_ABORTED == _spi_cb_event[_cb_event_idx])
-  {
-      end();
-  }
+        for (auto const start = millis();
+            (SPI_EVENT_TRANSFER_COMPLETE != _spi_cb_event[_cb_event_idx]) && (millis() - start < 1000); )
+        {
+            __NOP();
+        }
+        if (SPI_EVENT_TRANSFER_ABORTED == _spi_cb_event[_cb_event_idx])
+        {
+            end();
+        }
+    }
+    else
+    {
+        if (buf != NULL) {
```

Comment:
This can be checked at the very top and used to loose this function early. Saves one level of indentation for if/else, i.e.

```C++
void ArduinoSPI::transfer(void *buf, size_t count)
{
  if (!buf) return;
  ...
```

Reply:
Done, I did not do this before since it was not there for the FSP library which hopefully means that the FSP library is doing the check.

Reply:
I believe I suggested a "return-early" statement, as in my demo above. https://github.com/arduino/ArduinoCore-renesas/pull/45/commits/0cc34e0d87009cc772fc13c49953be49dda7a699 does nothing to decrease the level of indentation and if/else statements in the function. Please fix this.

The FSP code does handle a null [imho](https://github.com/renesas/fsp/blob/master/ra/fsp/src/r_sci_spi/r_sci_spi.c#L694-L764).

Reply:
Ok, done, this was intentially though as coding guidelines generally only want one return at the end of the function.
While applying even less strict rulesets against the codebase will result in a huge amount of violations, I am trying to minimize the amount of violations in my contribution. 

For example BARR-C: 2018:
"6.2.c. It is a preferred practice that all functions shall have just one exit point and it shall be via a return at the bottom of the function."

From the only demo of pc-lint plus I get for MISRA C++ 2018:
note 904: return statement before end of function 'transfer' [MISRA C++ Rule 6-6-5]

Well now, the purpose should be clear in this case, there still is only one return, these rules are not mandatory and this does indeed remove one level of indent.